chiselFrontend/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -186,6 +187,16 @@ private[chisel3] object Builder {
   def whenDepth_=(target: Int): Unit = {
     dynamicContext.whenDepth = target
   }
+  def currentClockAndReset: Option[ClockAndReset] = dynamicContext.currentClockAndReset
+  def currentClockAndReset_=(target: Option[ClockAndReset]): Unit = {
+    dynamicContext.currentClockAndReset = target
+  }
+  def forcedClockAndReset: ClockAndReset = currentClockAndReset match {
+    case Some(clockAndReset) => clockAndReset
+    case None => throwException("Error: No implicit clock and reset.")
+  }
+  def forcedClock: Clock = forcedClockAndReset.clock
+  def forcedReset: Bool = forcedClockAndReset.reset
```

Comment:
I was mirroring the `currentModule` and `forcedModule`, but my interpretation is that `currentClockAndReset` is an `Option` because you can be outside of a clock and reset scope (esp. once we have `RawModule`). `forcedClock` and `forcedReset` are documenting their use that you must have a clock and a reset in scope. It also provides a nice shorthand to just get the current Clock or Reset with the check built-in